converter.go

```diff
@@ -255,35 +256,67 @@ func arrowToValue(destcol *[]snowflakeValue, srcColumnMeta execResponseRowType,
 
 	switch strings.ToUpper(srcColumnMeta.Type) {
 	case "FIXED":
-		for i, int8 := range array.NewInt8Data(data).Int8Values() {
-			if !srcValue.IsNull(i) {
-				if srcColumnMeta.Scale == 0 {
-					(*destcol)[i] = int8
-				} else {
-					(*destcol)[i] = float64(int8) / math.Pow10(int(srcColumnMeta.Scale))
+		switch srcValue.DataType().ID() {
+		case arrow.DECIMAL:
+			// TODO
+		case arrow.INT64:
+			for i, val := range array.NewInt64Data(data).Int64Values() {
+				if !srcValue.IsNull(i) {
+					if srcColumnMeta.Scale == 0 {
+						(*destcol)[i] = val
+					} else {
+						r := big.NewRat(val, int64(math.Pow10(int(srcColumnMeta.Scale))))
+						(*destcol)[i] = r
+					}
+				}
+			}
+		case arrow.INT32:
+			for i, val := range array.NewInt32Data(data).Int32Values() {
+				if !srcValue.IsNull(i) {
+					if srcColumnMeta.Scale == 0 {
+						(*destcol)[i] = int64(val)
+					} else {
+						r := big.NewRat(int64(val), int64(math.Pow10(int(srcColumnMeta.Scale))))
+						(*destcol)[i] = r
+					}
+				}
+			}
+		case arrow.INT16:
+			for i, val := range array.NewInt16Data(data).Int16Values() {
+				if !srcValue.IsNull(i) {
+					if srcColumnMeta.Scale == 0 {
+						(*destcol)[i] = int64(val)
+					} else {
+						r := big.NewRat(int64(val), int64(math.Pow10(int(srcColumnMeta.Scale))))
+						(*destcol)[i] = r
+					}
+				}
+			}
+		case arrow.INT8:
+			for i, val := range array.NewInt8Data(data).Int8Values() {
+				if !srcValue.IsNull(i) {
+					if srcColumnMeta.Scale == 0 {
+						(*destcol)[i] = int64(val)
+					} else {
+						r := big.NewRat(int64(val), int64(math.Pow10(int(srcColumnMeta.Scale))))
+						(*destcol)[i] = r
+					}
 				}
-			} else {
-				(*destcol)[i] = nil
 			}
-
 		}
 		return err
 	case "BOOLEAN":
 		boolData := array.NewBooleanData(data)
 		for i := range *destcol {
 			if !srcValue.IsNull(i) {
 				(*destcol)[i] = boolData.Value(i)
-			} else {
-				(*destcol)[i] = nil
```

Comment:
does this mean the destcol is already nil initialized?

Reply:
Yes